src/ngx_base_fetch.cc

```diff
@@ -120,13 +120,15 @@ void NgxBaseFetch::ReadCallback(const ps_event_data& data) {
   NgxBaseFetch* base_fetch = reinterpret_cast<NgxBaseFetch*>(data.sender);
   ngx_http_request_t* r = base_fetch->request();
   bool detached = base_fetch->detached();
-  const char* type = BaseFetchTypeToCStr(base_fetch->base_fetch_type_);
-  int refcount = base_fetch->DecrementRefCount();
 
   #if (NGX_DEBUG)
+  const char* type = BaseFetchTypeToCStr(base_fetch->base_fetch_type_);
+  int refcount = base_fetch->DecrementRefCount();
   ngx_log_error(NGX_LOG_DEBUG, ngx_cycle->log, 0,
      "pagespeed [%p] event: %c. bf:%p (%s) - refcnt:%d - det: %c", r,
      data.type, base_fetch, type, refcount, detached ? 'Y': 'N');
+  #else
+  int refcount = base_fetch->DecrementRefCount();
```

Comment:
Can't we just move `DecrementRefCount()` after the NGX_DEBUG section and avoid the  duplication?